doc/tutorials/cne/cne.txt

```diff
@@ -0,0 +1,233 @@
+/*!
+
+@file cne.txt
+@author Kartik Nighania
+@brief Tutorial on how to use the CNE optimizer class.
+
+@page cnetutorial CNE Optimizer tutorial (mlpack_cne)
+
+@section intro_cnetut Introduction
+
+Conventional Neural Evolution (CNE) is a class of evolutionary algorithms
+focused on dealing with fixed topology networks. This class implements this
+algorithm as an optimization technique to converge a given function to minima.
+
+The algorithm works by creating a fixed number of candidates. The candidates
+are simply the network parameters that are untrained. Given the train and 
+test set. Each candidate is tested upon the training set or 1 epoch and then
+a fitness score is assigned to it. Given the selection percentage of best
+candidates by the user, for a single generation that many percentage of 
+candidates are selected for the next generation and the rest are removed.
+The selected candidates for a particular generation then become the parent 
+for the next generation and evolution takes place.
+
+@section toc_cnetut Table of Contents
+
+A list of all the sections this tutorial contains.
+
+ - \ref intro_cnetut
+ - \ref toc_cnetut
+ - \ref cne_cnetut
+   - \ref cne_ex1_cnetut
+   - \ref cne_ex2_cnetut
+   - \ref cne_ex3_cnetut
+ - \ref further_doc_cnetut
+
+@section cne_cnetut The CNE optimizer class
+
+The CNE class is a simple implementation of cne optimizer to converge a given 
+neural network.
+
+Using the cne class is very simple and can be divided into 3 simple steps- 
+
+1) The CNE object is made in which the constructor requires 7 input parameters.
+   The default values and detailed explaination have been discussed in a seperate
+   section below.
+
+@code
+CNE opt(const size_t populationSize,
+        const size_t maxGenerations,
+        const double mutationProb,
+        const double mutationSize,
+        const double selectPercent,
+        const double finalValue,
+        const double fitnessHist);
+@endcode
+
+2) Making a neural network model and giving CNE as an optimizer to train the model.
+   For our test, we will be using a feed forward network or vanilla network from the
+   artificial neural network class.
+
+
+3) The trained model can then be used by calling the predict method.
+
+@code
+void Predict(arma::mat predictors, arma::mat& results);
+@endcode
+
+Given the data to predict in armadillo matrix format. Matrix result is modified
+and the output of prediction is stored in it.
+
+@subsection cne_ex1_cnetut The constructor parameters.
+
+@code
+CNE(const size_t populationSize = 15,
```

Comment:
We should use the same default values here as for the definition, e.g. 500 for the ``populationSize`` instead of 15.